lib/diyepw/diyepw/convert_noaa_isd_lite_to_amy_epw_files.py

```diff
@@ -0,0 +1,172 @@
+            #    case
+            amy_df_s = _clean_noaa_df(amy_df_s)
+
+            # Grab the appropriate number of time steps for the subsequent year.
+            amy_df_s = amy_df_s.head(abs_time_steps)
+
+            # Append the NOAA dataframe for the subsequent year to the dataframe for the year of interest.
+            amy_df = amy_df.append(amy_df_s)
+
+            # Shift the timestamp (index) to match the time zone of the WMO station.
+            amy_df = amy_df.shift(periods=tz_shift, freq='H')
+
+            # Remove time steps that aren't applicable to the year of interest
+            amy_df = amy_df[amy_df.index >= start_timestamp]
+
+            _handle_missing_values(
+                amy_df,
+                step=_pd.Timedelta("1h"),
+                max_to_interpolate=max_records_to_interpolate,
+                max_to_impute=max_records_to_impute,
+                imputation_range=_pd.Timedelta("2w"),
+                imputation_step=_pd.Timedelta("1d"),
+                missing_values=[_np.nan, -9999.]
+            )
+
+            # Initialize new column for station pressure (not strictly necessary)
+            amy_df['Station_Pressure'] = None
+
+            # Convert sea level pressure in NOAA df to atmospheric station pressure in Pa.
+            for index in amy_df.index:
+                stp = _convert_sea_level_pressure_to_station_pressure(amy_df['Sea_Level_Pressure'][index], tmy.elevation)
+                amy_df.loc[index, 'Station_Pressure'] = stp
+
+            # Change observation values to the values taken from the AMY data
+            tmy.set('year', year)
+            tmy.set('Tdb', [i / 10 for i in amy_df['Air_Temperature']]) # Convert AMY value to degrees C
+            tmy.set('Tdew', [i / 10 for i in amy_df['Dew_Point_Temperature']]) # Convert AMY value to degrees C
+            tmy.set('Patm', amy_df['Station_Pressure'])
+            tmy.set('Wdir', amy_df['Wind_Direction'])
+            tmy.set('Wspeed', [i / 10 for i in amy_df['Wind_Speed']]) # Convert AMY value to m/sec
+
+            # Check for violations of EPW file standards
+            epw_rule_violations = tmy.validate_against_epw_rules()
+            if len(epw_rule_violations) > 0:
+                raise Exception("EPW validation failed:\n" + "\n".join(epw_rule_violations))
+
+            # Write new EPW file if no validation errors were found.
+            amy_epw_file_name = f"{tmy.country}_{tmy.state}_{tmy.city}.{tmy.station_number}_AMY_{year}.epw"
+            amy_epw_file_name = amy_epw_file_name.replace(" ", "-")
+            amy_epw_file_path = _os.path.join(output_dir, amy_epw_file_name)
+            tmy.write_epw(amy_epw_file_path)
+            amy_epw_file_paths.append(amy_epw_file_path)
+
+        except Exception as e:
+            errors = errors.append({"file": amy_file_path, "error": str(e)}, ignore_index=True)
+            print('Problem processing ' + amy_file_path + ': ' + str(e))
```

Comment:
This should be done using a logger and not `print`

Reply:
This is resolved in the csv_based_script branch; all prints in that branch have been removed or converted to logger calls.